pkg/apis/serving/v1beta1/inference_service_defaults.go

```diff
@@ -145,6 +145,14 @@ func (isvc *InferenceService) setPredictorModelDefaults() {
 	case isvc.Spec.Predictor.Paddle != nil:
 		isvc.assignPaddleRuntime()
 	}
+
+	if isvc.Spec.Predictor.Model != nil && isvc.Spec.Predictor.Model.ProtocolVersion == nil {
```

Comment:
Wondering why this logic is not executed in SetTritonDefaults?

Reply:
That function is only called after selecting the runtime. https://github.com/kserve/kserve/blob/294a10495b6b5cda9c64d3e1573b60aec62aceb9/pkg/controller/v1beta1/inferenceservice/components/predictor.go#L156